packages/Search/src/details/DTK_DetailsTreeConstruction_def.hpp

```diff
@@ -103,10 +103,13 @@ class GenerateHierarchyFunctor
     GenerateHierarchyFunctor(
         Kokkos::View<unsigned int *, DeviceType> sorted_morton_codes,
         Kokkos::View<Node *, DeviceType> leaf_nodes,
-        Kokkos::View<Node *, DeviceType> internal_nodes )
+        Kokkos::View<Node *, DeviceType> internal_nodes,
+        Kokkos::View<int *, DeviceType> parents )
         : _sorted_morton_codes( sorted_morton_codes )
         , _leaf_nodes( leaf_nodes )
         , _internal_nodes( internal_nodes )
+        , _parents( parents )
+        , _shift( internal_nodes.extent( 0 ) )
```

Comment:
Don't like the `_shift` name, it's not descriptive by itself. Maybe call it `_leafs_offset`, or something similar?